elbepack/xmlpreprocess.py

```diff
@@ -143,6 +143,33 @@ def preprocess_initvm_ports(xml):
             forward.getparent().remove(forward)
 
 
+def preprocess_pkg_pinning(xml):
+    """Do search and replace on pkg attributes, replacing 'pin' with 'release-name'."""
+
+    errors = False
+    for pkg in xml.iterfind('//target/pkg-list/pkg'):
+        if 'pin' in pkg.attrib.keys():
+            logging.warning('Attribute pin= for element <pkg> is deprecated. ' \
+                'Use release-name= instead.')
+
+            pkg.attrib['release-name'] = pkg.attrib['pin']
```

Comment:
Please also test that 'release-name' is not already set.

Reply:
done